### PR TITLE
[SPARK-41155][SQL] Add error message to SchemaColumnConvertNotSupportedException

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

This patch adds error message to `SchemaColumnConvertNotSupportedException`.

### Why are the changes needed?

Just found the fact that `SchemaColumnConvertNotSupportedException` doesn't have any error message is annoying for debugging.

In stack trace, we only see `SchemaColumnConvertNotSupportedException` but don't know what column is wrong.

After this change, we should be able to see it, e.g.,

```
org.apache.spark.sql.execution.datasources.SchemaColumnConvertNotSupportedException: column: [_1], physicalType: INT32, logicalType: bigint
```

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by viirya** [comment]
cc @dongjoon-hyun @sunchao

**Review by dongjoon-hyun — APPROVED** [review]
+1,LGTM.

**Comment by viirya** [comment]
Thank you @dongjoon-hyun !

**Review by sunchao — APPROVED** [review]
LGTM too

**Comment by viirya** [comment]
Thank you @sunchao !

**Review by wangyum — APPROVED** [review]

File: sql/core/src/main/java/org/apache/spark/sql/execution/datasources/SchemaColumnConvertNotSupportedException.java [hunk 1]
```diff
@@ -54,7 +54,8 @@ public SchemaColumnConvertNotSupportedException(
       String column,
       String physicalType,
       String logicalType) {
-    super();
+    super("column: " + column + ", physicalType: " + physicalType +
+            ", logicalType: " + logicalType);
```
Comment (dongjoon-hyun) [inline sql/core/src/main/java/org/apache/spark/sql/execution/datasources/SchemaColumnConvertNotSupportedException.java:58]: Is the indentation correct, @viirya ?

Comment (viirya) [reply sql/core/src/main/java/org/apache/spark/sql/execution/datasources/SchemaColumnConvertNotSupportedException.java]: Just fixed it. Thanks for reminder.

**Comment by viirya** [comment]
Thanks @wangyum too!

Previous run passed all tests. Only Java linter failed. I will merge this once Java Linter passes.

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Comment by viirya** [comment]
Linter was passed. There are more than one runs passed all tests. So I'm going to merge this. Thanks @dongjoon-hyun @sunchao @wangyum @LuciferYang

**Comment by dongjoon-hyun** [comment]
Sure Go ahead~

**Comment by viirya** [comment]
Merged. Thanks.

**Pull request closed without merging**